### PR TITLE
Optimization for Gdn_Cache class to store statically prefix value.

### PR DESCRIPTION
**Pull request by alex-mtl**
Optimization for Gdn_Cache class to store statically prefix value.

Relates to : #7689 and #7734
Fixes: #7788

### CODE REVIEW

File: library/core/class.cache.php [hunk 1]
```diff
@@ -564,9 +570,14 @@ protected function fallback($key, $options) {
 
     public function getPrefix($forcePrefix = null, $withRevision = true) {
         static $configPrefix = false;
+        $cKey = ($forcePrefix ?? 'null:').($withRevision ? 'true' : 'false');
+        if (key_exists($cKey, self::$cConfig)) {
```
Comment (charrondev) [inline library/core/class.cache.php]: Shouldn't this be `array_key_exists()`? Looks like this an alias from PHP <5 that's still around.

Comment (alex-mtl) [reply library/core/class.cache.php]: Done.

**Comment by initvector** [comment]
I'm putting a hold tag on this PR until the internal discussion has been resolved. Please report back here with whatever is decided.

**Comment by kaecyra** [comment]
In my opinion we should not even think about merging a change to something this deep in Vanilla without first implementing unit tests for how the methods USED to behave. Otherwise we're no better than animals and just rolling the dice on a regression. With the potential for really screwing things up on deploy.

**Comment by charrondev** [comment]
Closing until those unit tests are present.

**Pull request closed without merging**